examples/webgl_loader_mmd.html

```diff
@@ -160,23 +172,76 @@
 
 				if( mesh ) {
 
-					mixer.update( clock.getDelta() );
-
-					/* Note: ikSolver.resetLinks() is workaround for r74dev.
-					 *       THREE.PropertyMixer.apply() seems to have buffer cache
-					 *       when mixer.update() is called.
-					 *       ikSolver.update() change bone position/quaternion
-					 *       without mixer.update() then buffer cache will be inconsistent.
-					 *       ikSolver.resetLinks() explicitly resets IK link bones quaternion and
-					 *       tries to avoid issues caused by this inconsistent.
-					 */
-					ikSolver.resetLinks();
+					var delta = clock.getDelta();
+					mixer.update( delta );
+
+					// see below for what this's for
+					backupBones( mesh );
+
 					ikSolver.update();
+					physics.update( delta );
 
 				}
 
 				renderer.render( scene, camera );
 
+				if ( mesh ) {
+
+					// see below for what this's for
+					restoreBones( mesh );
+
+				}
+
+			}
+
+			/*
+			 * Note: These following three functions are workaround for r74dev.
+			 *       THREE.PropertyMixer.apply() seems to save values into buffer cache
```

Comment:
> > When the animation system binds to a property it must not be changed by other means. It just pushes out a defined state, so even with the workaround it will overwrite previous changes?!

> In MMD system, IK and Physics calculate the final bone position/quaternion from pure skinning result in each frame.
> ```javascript
> var delta = clock.getDelta();
> ikSolver.update(); // directly(without bindings) changing bone position/quaternion
> physics.update( delta );  // directly(without bindings) changing bone position/quaternion
> mixer.update( delta ); // mixer is an instance of THREE.AnimationMixer
> ```

OK, so this code is a little mixed up and `mixer.update(dt)` actually runs first, then IK & Physics, right?

I think a both cleaner and faster workaround might be to have a second bones structure and let the animation system emit its state there. Then copy it for IK & Physics. 

How do these components address the bones? Are they reusable? Would there be a way to make them operate on the `PropertyMixers` and turn them into some kind of mixer plugins?